lib/libzfs/libzfs_util.c

```diff
@@ -891,6 +891,8 @@ libzfs_run_process_impl(const char *path, char *argv[], char *env[], int flags,
 	 */
 	if (lines != NULL && pipe2(link, O_NONBLOCK | O_CLOEXEC) == -1)
 		return (-EPIPE);
+	if (access(path, X_OK))
```

Comment:
`R_OK | X_OK`

file can be executable but not readable in a degenerate case

Reply:
It's binary, you don't need to read it.
root@8ac66c5b4fb6:/# ls -l /usr/sbin/exportfs
--wx--x--x 1 root root 68056 May 24  2021 /usr/sbin/exportfs
root@8ac66c5b4fb6:/# !$ -h             
/usr/sbin/exportfs -h
usage: exportfs [-adfhioruvs] [host:/path]
root@8ac66c5b4fb6:/# 

I can ammend commit msg. What would you like me to include in it?

Reply:
If, by any chance, the file we would like to execute is a script with a shebang, `execve()` would fail as it requires read permission to succeed.

Would
`libzfs_run_process_impl(): check if file is executable before execve()`
work for you?

Reply:
GitHub ate my previous comment...

In case the executed binary is a script with the shebang, `execve()` will fail - read permission would be necessary to proceed (check it yourself).

Would `libzfs_run_process_impl(): check if file is executable before execve()` work for you?

Reply:
In implemented case, file being run is binary. One can have executable file without read permissions.
Commit message set as requested.

Reply:
In your specific case it's an ELF binary. Others might have scripts, e. g. shell wrappers, which still result in an error from this PR perspective.
Changes in libraries should cover general cases, not be specific to the reproduction environment.